src/hooks/useDirectoryContent.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useSelector } from 'react-redux';
+import React, { useRef, useEffect, useCallback, useState } from 'react';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { fetchElementsInfos } from '../utils/rest-api';
+import { UUID } from 'crypto';
+import { IElement, IElementMetadata, ReduxState } from '../redux/reducer.type';
+
+export const useDirectoryContent = (
+    setIsMissingDataAfterDirChange: React.Dispatch<
+        React.SetStateAction<boolean>
+    >
+) => {
+    const currentChildren = useSelector(
+        (state: ReduxState) => state.currentChildren
+    );
+    const [childrenMetadata, setChildrenMetadata] = useState<
+        Record<UUID, IElementMetadata>
+    >({});
+    const { snackError } = useSnackMessage();
+
+    const [rows, setRows] = useState<IElement[]>(currentChildren);
```

Comment:
what is the point of this state since it seems to be equals to currentChildren at all time ?

Reply:
It was initially to follow documentation guidelines but looking back at it it seems unnecessary bloating, I'll directly return currentChildren I think